### PR TITLE
[AIRFLOW-3245] fix list processing in resolve_template_files

### PR DESCRIPTION
**Pull request by ms32035**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-3245

### Description

- [X] resolve_template_files does not process template field value if it is a list

### Tests

- [x] Looks like currently the function isn't directly tested. If needed, please advise where should I add a test case

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- [X] Passes `flake8`

### CODE REVIEW

**Comment by kaxil** [comment]
Can you please add tests for this?

**Comment by ms32035** [comment]
@kaxil, can you direct me where, cause I couldn't find any tests related to generating templates from files

**Comment by ms32035** [comment]
@kaxil tests added and build is passing. Please merge

File: tests/models.py [hunk 1]
```diff
@@ -454,6 +454,48 @@ def jinja_udf(name):
         result = task.render_template('', "{{ 'world' | hello}}", dict())
         self.assertEqual(result, 'Hello world')
 
+    def test_resolve_template_files_value(self):
+
+        f = NamedTemporaryFile(suffix='.template')
```
Comment (Fokko) [inline tests/models.py]: Please use with comprehension here

File: tests/models.py [hunk 2]
```diff
@@ -454,6 +454,48 @@ def jinja_udf(name):
...
+
+        task.test_field = template_file
+        task.template_fields = ('test_field',)
+        task.template_ext = ('.template',)
+        task.resolve_template_files()
+        self.assertEqual(task.test_field, '{{ ds }}')
+
+    def test_resolve_template_files_list(self):
+
+        f = NamedTemporaryFile(suffix='.template')
```
Comment (Fokko) [inline tests/models.py]: Please use with comprehension here

**Pull request merged**